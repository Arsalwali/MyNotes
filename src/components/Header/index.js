import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { colors, metrics, iconNames, iconTypes } from '../../theme';
import { Icon } from 'react-native-elements';

const styles = StyleSheet.create({
  container: {
    height: 60,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: colors.white,
    elevation: 5,
  },
  textStyle: {
    fontSize: metrics.header,
    fontWeight: 'bold',
    color: colors.primary1,
  },
  rightIconContainer: {
    position: 'absolute',
    right: 0,
    marginRight: 10,
  },
});


function Header(props) {
  return (
    <View style={styles.container}>
      <Text style={styles.textStyle}>{props.title}</Text>
      { props.rightIcon && 
        <View style={styles.rightIconContainer}>
          <Icon
            name={props.rightIcon}
            type={iconTypes.MaterialIcons}
            size={metrics.iconSize}
            color={colors.iconColor}
            onPress={props.onIconPress}
          />
        </View>
      }
    </View>
  );
}

export default Header;