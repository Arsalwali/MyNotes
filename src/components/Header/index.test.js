import React from 'react';
import { shallow } from 'enzyme';
import Header from './';

describe('Header component', () => {
  it('should render properly without right icon', () => {
    const props = {
      title: 'My Notes',
    };
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render properly with right icon', () => {
    const props = {
      title: 'My Notes',
      rightIcon: 'add',
    };
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
