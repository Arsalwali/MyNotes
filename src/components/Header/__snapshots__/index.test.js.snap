// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render properly with right icon 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "elevation": 5,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#517fa4",
        "fontSize": 27.5,
        "fontWeight": "bold",
      }
    }
  >
    My Notes
  </Text>
  <View
    style={
      Object {
        "marginRight": 10,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Icon
      color="#517fa4"
      name="add"
      raised={false}
      reverse={false}
      reverseColor="white"
      size={28}
      type="materialicon"
      underlayColor="white"
    />
  </View>
</View>
`;

exports[`Header component should render properly without right icon 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "elevation": 5,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#517fa4",
        "fontSize": 27.5,
        "fontWeight": "bold",
      }
    }
  >
    My Notes
  </Text>
</View>
`;
